plugins/modules/nim.py

```diff
@@ -1067,12 +1067,12 @@ def nim_maintenance(module, params):
 
     for target in results['targets']:
         module.log('NIM - perform maintenance operation for client {0}'.format(target))
-        results['meta'][target] = {'messages': []}  # first time init
+        results['meta']['messages'] = {'messages': []}  # first time init
         results['status'][target] = ''  # first time init
 
         if target in results['nim_node']['vios']:
             msg = 'maintenance operation is not supported on VIOS.'
-            results['meta'][target]['messages'].append(msg)
+            results['meta']['messages'].append(msg)
```

Comment:
I am a bit confused with this change.
Previously we save the msg, cmd, rc, stdout, stderr
for each target (if there are multiple targets)

with the current change, we will only save the results of
the last target processed

Reply:
Hi, has this question been resolved? Thanks

Reply:
I think you did not check the define of results,  meta does not contain target,  it was commented out. so meta only can save message

  results = dict(
        changed=False,
        msg='',
        status={},
        stdout='',
        stderr='',
        meta={'messages': []},
        # meta structure will be updated as follow:
        # meta={
        #   target_name:{
        #       'messages': [],     detail execution messages
        #       'cmd': '',
        #       'rc': '',
        #       'stdout': '',
        #       'stderr': '',
        #   }
        # }
        nim_node={},
    )

Reply:
the define is a bit confusing.
i think the person who wrote the comments meant to imply
that over the course the of execution the meta structure
will be updated into the struct of the commented section
of the define

Reply:
python dictionaries are kinda weird, they are quite fluid unlike the structs in C.
although in the define, it only had one key `message`

if you write `meta['target'] = {...}` it will actually create a new entry in the dictionary

Reply:
The reason of I am changing code is because it caused the message missing in the log, even if it create a new entry, it still won't give right info... no matter what comments are given, we should always keep code and define consistent.